src/fizzbuzz/domain/FizzBuzz-reducer.ts

```diff
@@ -0,0 +1,18 @@
+import { FizzBuzz } from './FizzBuzz';
+
+export class FizzBuzzReducer {
+  static reduce({ from, count }, reducer) {
+    const numbers = FizzBuzzReducer.toArray(from, count);
+    return numbers.reduce((acc, curr) => {
+      return acc + reducer() + new FizzBuzz(curr).say();
+    }, '');
+  }
+
+  private static toArray(from: number, count: number): Array<number> {
+    const array = [];
```

Comment:
functional way is something like `Array.from({ length: count}).map((_, index) => from + index)`